syntax_and_semantics/comments.md

```diff
@@ -1,6 +1,6 @@
 # Comments
 
-Comments start with the sharp (`#`) character. Only one-line comments are supported for now.
+Comments start with the number (`#`) sign. Only one-line comments are supported for now.
```

Comment:
I think it would even be better to move the `#` behind "number sign" instead of having it hanging strangely in between.

Reply:
Why not just "Comments start with `#`. [...]"

Reply:
For searchability it's good to write the name of the character, because searching for `#` is PITA.

Reply:
I tried googling for "crystal comments start with the sharp" and it doesn't find it. Besides that, if you don't know how to comment Crystal code you search for "crystal how to comment", I've got some difficulties figuring out a scenario where I'd be searching for "crystal sharp" or "crystal number sign" and wanting to get to this page as a result.

Reply:
I don't have a convincing example either (maybe something like `number sign in programming language`) but that's not really the point. Is there any benefit from *not* including the name?

Reply:
Agreed, why would I search for the character when I don't know how do to comments at all. Also there's way too many names for `#`, pound sign, hash, crosshatch, octothorpe (according to unicode anyway :D) ...

Reply:
If nothing else, it helps to clarify it is `#` and none of these similar looking - depending on the font - unicode characters `⋕⌗♯ⵌ井ꖛ﹟＃𐦕𝄰𝄱`.

Reply:
People that confuse it with those kinda deserve the pain if you ask me ... :P

Reply:
Quick poll: which of each of those chars is "number sign", which is "hash sign", which is "number sign"? I'll invite you a beer if all of us here get it right for all cases :D.

Reply:
I vote for drop the name entirely. If I heard "number sign" I wouldn't think `#`, but I would think `#` for "sharp" or "hash". Since the name is likely to be contentious, lets just use the specific character.

Reply:
In 38 years of speaking English, I have NEVER heard that referred to as a "sharp" outside of music class. The terms "number sign" or "pound sign" are pretty universal, so that's why I put "number sign."

Also, the Oxford English dictionary agrees that it's the number sign.
https://en.oxforddictionaries.com/definition/number_sign

Reply:
However, if you'd all rather just say "# character," then I'll update it to that instead.

Reply:
Maybe only musicians know it as sharp? :-)

Reply:
I thought most programmers would know about C# being pronounced C sharp.

Reply:
I thought most programmers have enough attention to detail to notice that the C sharp logo is a sharp, not a number sign. Although the two are similar, they're not the same.

Reply:
Snarkiness aside, the musical sharp ♯ is Unicode 266F, while the number sign # is Unicode 0023. Can we put this to rest now and move on to the best wording for this page?

Reply:
Lets just use the character with no name. Please.

Reply:
Can we get a vote on this? We've had a number of commenters here, so I'd like to make sure we've got a consensus before I update the PR.

Reply:
Thumbs up here if you want it to say:

Comments start with the (`#`) character.

Reply:
Or, if you want it to say:

Comments start with the number sign (`#`) .

Reply:
@HCLarsen we already have 2 votes from the core team on not using a name, surely that's enough.